hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -8838,6 +8838,10 @@ private Supplier<String> getLockReportInfoSupplier(String src, String dst,
       UserGroupInformation ugi = Server.getRemoteUser();
       String userName = ugi != null ? ugi.toString() : null;
       InetAddress addr = Server.getRemoteIp();
+      if (addr != null) {
```

Comment:
@virajjasani Thanks for your works. But I am concerned if it will involve more serious impact here because when hold lock for a long time and trigger hostname resolution here which is one IO inner lock. Thanks again.

Reply:
hmm, FSNamesystemLock#coarseLock (writeLock) is unlocked only after writing above log so we will incur extra latency for long lock holding thread. Btw, i think coarseLock writeLock can be unlocked before logging the info as well (by protecting `longestWriteLockHeldInfo` with either synchronizing or using another lock), but this is out of scope here so maybe sometime later.

shall we keep ip address only for long lock hold audit logs (similar to FSNamesystem audit logs)?

Reply:
Strong +1 to update TestFSNamesystemLockReport only. And move log out of lock could improve at next PR. Thanks.

Reply:
Sounds good, let me update the test. Thanks @Hexiaoqiao